src/lib/pickles/pickles.ml

```diff
@@ -1257,6 +1354,153 @@ let%test_module "test no side-loaded" =
       assert (
         Promise.block_on_async_exn (fun () ->
             Tree_proof.Proof.verify_promise Tree_proof.example ) )
+
+    module Tree_proof_return = struct
+      module Statement = No_recursion_return.Statement
+
+      type _ Snarky_backendless.Request.t +=
+        | Is_base_case : bool Snarky_backendless.Request.t
+
+      let handler (is_base_case : bool)
+          (Snarky_backendless.Request.With { request; respond }) =
+        match request with
+        | Is_base_case ->
+            respond (Provide is_base_case)
+        | _ ->
+            respond Unhandled
+
+      let tag, _, p, Provers.[ step ] =
+        Common.time "compile" (fun () ->
+            compile_promise
+              (module Statement)
+              (module Statement.Constant)
+              ~public_input:(Output Field.typ)
+              ~branches:(module Nat.N1)
+              ~max_proofs_verified:(module Nat.N2)
+              ~name:"blockchain-snark"
+              ~constraint_constants:
+                (* Dummy values *)
+                { sub_windows_per_window = 0
+                ; ledger_depth = 0
+                ; work_delay = 0
+                ; block_window_duration_ms = 0
+                ; transaction_capacity = Log_2 0
+                ; pending_coinbase_depth = 0
+                ; coinbase_amount = Unsigned.UInt64.of_int 0
+                ; supercharged_coinbase_factor = 0
+                ; account_creation_fee = Unsigned.UInt64.of_int 0
+                ; fork = None
+                }
+              ~choices:(fun ~self ->
+                [ { identifier = "main"
+                  ; prevs = [ No_recursion_return.tag; self ]
+                  ; main =
+                      (fun [ _; prev ] () ->
+                        let is_base_case =
+                          exists Boolean.typ ~request:(fun () -> Is_base_case)
```

Comment:
👍🏻 for using `handler` / `~request` in this example. this actually taught me what handlers are all about :D i wonder whether I should use one for snarkyjs to pass in private inputs, would enable me to remove some global variable hacks